test/games/strategy/ui/SwingComponentsTest.java

```diff
@@ -29,4 +31,20 @@ public void testAppendExtensionIfAbsent_ShouldHandleExtensionThatStartsWithPerio
   public void testAppendExtensionIfAbsent_ShouldUseCaseInsensitiveComparisonForExtension() {
     assertThat(appendExtensionIfAbsent(new File("path/file.bBb"), "BbB"), is(new File("path/file.bBb")));
   }
+
+  @Test
+  public void testExtensionWithLeadingPeriod() {
+    assertThat(extensionWithLeadingPeriod("aaa"), is(".aaa"));
+    assertThat(extensionWithLeadingPeriod(".aaa"), is(".aaa"));
+    assertThat(extensionWithLeadingPeriod("aaa.aaa"), is(".aaa.aaa"));
+    assertThat(extensionWithLeadingPeriod(".aaa.aaa"), is(".aaa.aaa"));
+  }
+
+  @Test
+  public void testExtensionWithoutLeadingPeriod() {
+    assertThat(extensionWithoutLeadingPeriod("aaa"), is("aaa"));
```

Comment:
Done.  I also added a test case for the empty extension boundary condition (which didn't behave as I expected).  I'll proceed to rebase and squash -- it should be ready later today.

BTW, for my future reference, is there a preferred way in TripleA to write parameterized tests (other than doing what I did here by replicating the test multiple times)?  I did a quick search and didn't see anything obvious (e.g. the use of JUnit's `Parameterized` test runner).  I personally don't like the `Parameterized` runner that much, principally because of the need to extract one fixture per parameterized test.  I've used [JUnitParams](https://github.com/Pragmatists/JUnitParams) in the past and like it mostly because I can mix parameterized and non-parameterized tests in a single fixture.